wrangler/src/trust/trust.js

```diff
@@ -0,0 +1,271 @@
+ * @param oldturret - the address of the turret that is being removed from the source account
+ * @param newturret - the new turret to add to the source account
+ * @param functionhash - the hash of the function that is being healed
+ * @returns The transaction XDR, the signer public key, and the signature.
+ */
+export async function heal(sourceaccount, oldturret, newturret, functionhash) {
+    try{
+        /* check that the turret is not trying to heal itself */
+        if (newturret === TURRET_ADDRESS || oldturret === TURRET_ADDRESS)
+            throw ('A Turret may not add or remove itself to a control account')
+        /* gather source account signers */
+        const sourceaccountsigners = await getSignersOnAccount(sourceaccount)
+        /* find the local signer for given functionhash */
+        const localsigner = await getLocalFunctionSigner(functionhash)
+        /* verify the local signer is a signer on the source account */
+        if(!sourceaccountsigners.includes(localsigner))
+            throw (`This turret is not a signer on ${functionhash}, what are you doing?`)
+        /* gather the signer/turret managedata entries from the source account */
+        const sourceaccountmanagedatasigners = await getSourceAccountSignerDataEntries(sourceaccount)
+        /* check that the manage data does not yet include the new turret */
+        if (sourceaccountmanagedatasigners.turret.includes(newturret))
+            throw (`The New turret is already listed in the managedata entries for account`)
+        /* check that the oldturret is listed on the source account managedata entries */
+        if (!sourceaccountmanagedatasigners.turret.includes(oldturret))
+            throw (`The old turret is not listed as a signer in the manage data entries`)
+        /* get the signer address to remove from the managedata entries on the source account */
+        const signertoremoveindex = sourceaccountmanagedatasigners.turret.findIndex(checkaddr);
+        const removesigner = sourceaccountmanagedatasigners.signer[signertoremoveindex]
+        function checkaddr(key) {
+            return key === oldturret;
+        }
+        /* get the new signer address from the new turret */
+        const newturretsigner = await getTxFunctionSigner(functionhash, newturret)
+        /* verify the new turret is not already a signer on the source account */
+        if (sourceaccountsigners.includes(newturretsigner))
+            throw (`The new turret signer ${newturretsigner} is already a signer on sourceaccount ${sourceaccount}`)
+        /* assign the new signer weight by getting the old signer weight */
+        const signerweight = await getSignerWeight(removesigner, sourceaccount)
+        /* check the local toml to validate the new turret is part of the trust quorem and the old turret is not.*/
+        const newtrust = await checkLocalQuorem(newturret)
+        if (!newtrust){
```

Comment:
running eslint or prettier through there will fix small style typos like this automatically

if you don't have that configured just lmk and I can push a style commit through on this branch

Reply:
I do have a eslint config but not on this version, I appreciate if you add one.